### DISCUSSION TITLE
Trivy scan shows old version of jar

### DISCUSSION THREAD

**Discussion by AlanTT1** [aquasecurity/trivy#1129] [Q&A]
Hi! I am running Trivy on a Java war file.

Firstly, when I started the scan using this command:

`trivy fs -s HIGH,CRITICAL -f template -t "@/media/sf_vmsharedfolder/html.tpl" -o fse-trivy-report.html mydir`

there is an warning message:

`
2021-07-19T12:39:49.372+0800	INFO	Detected OS: unknown

2021-07-19T12:39:49.377+0800	INFO	Number of PL dependency files: 1

2021-07-19T12:39:49.377+0800	INFO	Detecting jar vulnerabilities...

2021-07-19T12:39:49.463+0800	WARN	maven constraint error ([10.5-alpha0,10.5.3.0_1]): failed to new comparer: 2 errors occurred:

	* improper constraint: [10.5-alpha0,10.5.3.0_1]
	* 
	* improper requirements: []
`

After this, when I look at the output, it shows an old jar file com.fasterxml.jackson.core:jackson-databind and the installed version is 2.3.3. I have checked the war file by opening it up, there is no version 2.3.3 in jackson-core.

This was the old version of jackson-core and I have already upgraded it to version 2.12.3. 

I have tried trivy with --reset and -c to clear cache but it still gives the same result.

Please help. Thanks.

**Comment by davidygp** [comment]
Same issue. I've having it with version 2.4.0 :/
Has it been solved?

**Reply by DmitriyLewen** [reply]
Hello @davidygp 
Can you tell which version of Trivy you are using. And which jar are you scanning?

Regards, Dmitriy

**Reply by davidygp** [reply]
Hi Dmitriy,

Thanks for your quick reply!

I'm using Trivy 0.30.2.
**Anyways I think I've figured it out, don't think its an issue with Trivy.**

I'm scanning a docker image which shows that there is a jackson-databind v2.4.0 vulnerability. 
But couldn't find the jackson-databind v2.4.0 jar. It was another version, which I subsequently replaced with a newer version.

Turns out it was the htrace-core4-4.1.0-incubating.jar which had a jackson-databind v2.4.0 dependency.
(Could see it in the <unzipped .jar>/META-INF/maven/com.fasterxml.jackson.core/jackson-databind/pom.xml file.)

Removed the offending htrace-core4-4.1.0-incubating.jar and its no longer showing the jackson-databind v2.4.0 vulnerability.

**TL;DR - Its cause Trivy scans the pom.xml for the each .jar's dependencies and highlights those as well (?).**

Best Regards,
David Yam

**Reply by DmitriyLewen** [reply]
Hello @davidygp 

You are almost right.
Trivy takes all included pom.properties (not pom.xml) files and scans those dependencies.

Otherwise you are correct: `htrace-core4-4.1.0-incubating.jar` contains (`/META-INF/maven/com.fasterxml.jackson.core/jackson-databind/pom.properties`) pom.properties file with `jackson-databind v2.4.0`.

Regards, Dmitriy

**Reply by seanleblancicdtech** [reply]
Is there any way to disable this behavior of scanning pom.properties that are inside jars?

**Reply by wenlong-ihoment** [reply]
pom.properties文件里面的依赖项有问题，有修复方案么？

**Reply by DmitriyLewen** [reply]
There is currently no way to disable `pom.properties` scanning from jar files.

**Comment by wenlong-ihoment** [comment]
There is a problem with the dependencies in the pom.properties file. Is there a solution?

**Reply by DmitriyLewen** [reply]
Hey @wenlong-ihoment
Can you clarify what your problem is?
